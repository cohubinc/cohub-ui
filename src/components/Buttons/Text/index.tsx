import React, { PureComponent, CSSProperties } from "react";
import Blank from "../Blank";
import Color from "src/definitions/enums/Color";
import Typography from "src/components/Typography";
import { TBlankButtonProps } from "src/components/Buttons/Blank";

interface IProps {
  color?: Color;
  fontSize?: number | string;
  textStyle?: CSSProperties;
  block?: boolean;
}

type TProps = IProps & Omit<TBlankButtonProps, "color">;

export default class Text extends PureComponent<TProps> {
  static defaultProps: Partial<IProps> = {
    color: Color.iconGrey,
    fontSize: "12px"
  };

  render() {
    const {
      color,
      fontSize,
      children,
      textStyle,
      block,
      style,
      className = "",
      ...rest
    } = this.props;

    return (
      <Blank
        {...rest}
        className={`${className} p-05`}
        style={{ display: block ? "block" : undefined, ...style }}
      >
        <Typography
          uppercase
          color={color}
          weight={500}
          style={{
            fontSize,
            ...textStyle
          }}
          kerning={0.07}
        >
          {children}
        </Typography>
      </Blank>
    );
  }
}
