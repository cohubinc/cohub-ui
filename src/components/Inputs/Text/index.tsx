import React from "react";
import { FieldRenderProps } from "react-final-form";

import TInputElementProps from "../definitions/TInputElementProps";
import Base from "../Base";

type FieldProps = FieldRenderProps<string, HTMLInputElement>;

interface ITextInputProps {
  label?: string;
  labelPosition?: "inside" | "outside" | "intersect";
  appearance?: "contrast" | "inverted";
  msgPosition?: {
    bottom: number;
  };
  "data-qa"?: string;
  input?: Partial<FieldProps["input"]>;
  meta?: FieldProps["meta"];
}

export type TTextInputProps = ITextInputProps &
  Omit<TInputElementProps, "onChange" | "value">;

export default function Text(props: TTextInputProps) {
  const {
    input,
    meta,
    style,
    className,
    msgPosition = {},
    "data-qa": dataQa,
    ...restProps
  } = props;

  const { touched, error } = meta || ({} as any);

  const showError = !!(touched && error);

  return (
    <div {...{ className, style: { width: "100%", ...style } }}>
      <Base {...input} {...restProps} error={showError} data-qa={dataQa} />
    </div>
  );
}
