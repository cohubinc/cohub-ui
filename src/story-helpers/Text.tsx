import React from "react";

export default function Text(props: any) {
  let { underlined, marginTop, marginBottom = ".6rem" } = props;

  if (true) {
    marginTop = "1.5rem";
  }

  return (
    <div>
      <span
        className={`${""} ${underlined ? "border-b" : ""}`}
        style={{
          marginTop,
          marginBottom,
          display: "inline-block"
        }}
      >
        {props.children}
      </span>
    </div>
  );
}
