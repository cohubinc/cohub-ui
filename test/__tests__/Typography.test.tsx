import React from "react";
import { mount } from "enzyme";

import Text from "src/components/Typography";

describe("Typography", () => {
  it("renders Regular Text without crashing", async () => {
    const wrapper = mount(<Text>Some Text</Text>);

    expect(wrapper).toMatchSnapshot();
  });

  it("renders Text.SuperTitle without crashing", async () => {
    const wrapper = mount(<Text.SuperTitle>Some Text</Text.SuperTitle>);

    expect(wrapper).toMatchSnapshot();
  });

  it("renders Text.Title without crashing", async () => {
    const wrapper = mount(<Text.Title>Some Text</Text.Title>);

    expect(wrapper).toMatchSnapshot();
  });

  it("renders Text.Subtitle without crashing", async () => {
    const wrapper = mount(<Text.Subtitle>Some Text</Text.Subtitle>);

    expect(wrapper).toMatchSnapshot();
  });

  it("renders Text.HeadingLarge without crashing", async () => {
    const wrapper = mount(<Text.HeadingLarge>Some Text</Text.HeadingLarge>);

    expect(wrapper).toMatchSnapshot();
  });

  it("renders Text.HeadingSmall without crashing", async () => {
    const wrapper = mount(<Text.HeadingSmall>Some Text</Text.HeadingSmall>);

    expect(wrapper).toMatchSnapshot();
  });

  it("renders Text.HeadingTiny without crashing", async () => {
    const wrapper = mount(<Text.HeadingTiny>Some Text</Text.HeadingTiny>);

    expect(wrapper).toMatchSnapshot();
  });

  it("renders Text.Large without crashing", async () => {
    const wrapper = mount(<Text.Large>Some Text</Text.Large>);

    expect(wrapper).toMatchSnapshot();
  });

  it("renders Text.Small without crashing", async () => {
    const wrapper = mount(<Text.Small>Some Text</Text.Small>);

    expect(wrapper).toMatchSnapshot();
  });

  it("renders Text.Tiny without crashing", async () => {
    const wrapper = mount(<Text.Tiny>Some Text</Text.Tiny>);

    expect(wrapper).toMatchSnapshot();
  });

  it("renders bold Text without crashing", async () => {
    const wrapper = mount(<Text weight={500}>Some Text</Text>);

    expect(wrapper).toMatchSnapshot();
  });

  it("renders muted Text without crashing", async () => {
    const wrapper = mount(<Text muted>Some Text</Text>);

    expect(wrapper).toMatchSnapshot();
  });

  it("renders inverted Text without crashing", async () => {
    const wrapper = mount(<Text inverted>Some Text</Text>);

    expect(wrapper).toMatchSnapshot();
  });
});
