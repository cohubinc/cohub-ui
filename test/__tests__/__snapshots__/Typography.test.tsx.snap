// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography renders Regular Text without crashing 1`] = `
<Typography>
  <Component>
    <span
      className=""
      data-qa="text"
      style={
        Object {
          "color": "hsl(0, 2%, 22%)",
          "fontFamily": "Inter",
          "fontSize": "14px",
          "fontWeight": 400,
          "textAlign": "initial",
          "textTransform": "initial",
        }
      }
    >
      Some Text
    </span>
  </Component>
</Typography>
`;

exports[`Typography renders Text.HeadingLarge without crashing 1`] = `
<Component>
  <span
    className=""
    data-qa="heading-large"
    style={
      Object {
        "color": "hsl(0, 2%, 22%)",
        "display": "block",
        "fontFamily": "Inter",
        "fontSize": 24,
        "fontWeight": 400,
        "textAlign": "initial",
        "textTransform": "initial",
      }
    }
  >
    Some Text
  </span>
</Component>
`;

exports[`Typography renders Text.HeadingSmall without crashing 1`] = `
<Component>
  <span
    className=""
    data-qa="heading-small"
    style={
      Object {
        "color": "hsl(0, 2%, 22%)",
        "display": "block",
        "fontFamily": "Inter",
        "fontSize": 20,
        "fontWeight": 400,
        "textAlign": "initial",
        "textTransform": "initial",
      }
    }
  >
    Some Text
  </span>
</Component>
`;

exports[`Typography renders Text.HeadingTiny without crashing 1`] = `
<Component>
  <span
    className=""
    data-qa="heading-tiny"
    style={
      Object {
        "color": "hsl(0, 2%, 22%)",
        "display": "block",
        "fontFamily": "Inter",
        "fontSize": 18,
        "fontWeight": 400,
        "textAlign": "initial",
        "textTransform": "initial",
      }
    }
  >
    Some Text
  </span>
</Component>
`;

exports[`Typography renders Text.Large without crashing 1`] = `
<Component>
  <span
    className=""
    data-qa="text"
    style={
      Object {
        "color": "hsl(0, 2%, 22%)",
        "fontFamily": "Inter",
        "fontSize": 16,
        "fontWeight": 400,
        "textAlign": "initial",
        "textTransform": "initial",
      }
    }
  >
    Some Text
  </span>
</Component>
`;

exports[`Typography renders Text.Small without crashing 1`] = `
<Component>
  <span
    className=""
    data-qa="text"
    style={
      Object {
        "color": "hsl(0, 2%, 22%)",
        "fontFamily": "Inter",
        "fontSize": 12,
        "fontWeight": 400,
        "textAlign": "initial",
        "textTransform": "initial",
      }
    }
  >
    Some Text
  </span>
</Component>
`;

exports[`Typography renders Text.Subtitle without crashing 1`] = `
<Component>
  <span
    className=""
    data-qa="page-subtitle"
    style={
      Object {
        "color": "hsl(0, 2%, 22%)",
        "display": "block",
        "fontFamily": "Inter",
        "fontSize": 30,
        "fontWeight": 400,
        "textAlign": "initial",
        "textTransform": "initial",
      }
    }
  >
    Some Text
  </span>
</Component>
`;

exports[`Typography renders Text.SuperTitle without crashing 1`] = `
<Component>
  <span
    className=""
    data-qa="page-supertitle"
    style={
      Object {
        "color": "hsl(0, 2%, 22%)",
        "display": "block",
        "fontFamily": "Inter",
        "fontSize": 48,
        "fontWeight": 400,
        "lineHeight": "normal",
        "textAlign": "initial",
        "textTransform": "initial",
      }
    }
  >
    Some Text
  </span>
</Component>
`;

exports[`Typography renders Text.Tiny without crashing 1`] = `
<Component>
  <span
    className=""
    data-qa="text"
    style={
      Object {
        "color": "hsl(0, 2%, 22%)",
        "fontFamily": "Inter",
        "fontSize": 10.2,
        "fontWeight": 400,
        "textAlign": "initial",
        "textTransform": "initial",
      }
    }
  >
    Some Text
  </span>
</Component>
`;

exports[`Typography renders Text.Title without crashing 1`] = `
<Component>
  <span
    className=""
    data-qa="page-title"
    style={
      Object {
        "color": "hsl(0, 2%, 22%)",
        "display": "block",
        "fontFamily": "Inter",
        "fontSize": 36,
        "fontWeight": 400,
        "lineHeight": "normal",
        "textAlign": "initial",
        "textTransform": "initial",
      }
    }
  >
    Some Text
  </span>
</Component>
`;

exports[`Typography renders bold Text without crashing 1`] = `
<Typography
  weight={500}
>
  <Component
    weight={500}
  >
    <span
      className=""
      data-qa="text"
      style={
        Object {
          "color": "hsl(0, 2%, 22%)",
          "fontFamily": "Inter",
          "fontSize": "14px",
          "fontWeight": 500,
          "textAlign": "initial",
          "textTransform": "initial",
        }
      }
    >
      Some Text
    </span>
  </Component>
</Typography>
`;

exports[`Typography renders inverted Text without crashing 1`] = `
<Typography
  inverted={true}
>
  <Component
    inverted={true}
  >
    <span
      className=""
      data-qa="text"
      style={
        Object {
          "color": "hsl(0, 0%, 98%)",
          "fontFamily": "Inter",
          "fontSize": "14px",
          "fontWeight": 400,
          "textAlign": "initial",
          "textTransform": "initial",
        }
      }
    >
      Some Text
    </span>
  </Component>
</Typography>
`;

exports[`Typography renders muted Text without crashing 1`] = `
<Typography
  muted={true}
>
  <Component
    muted={true}
  >
    <span
      className=""
      data-qa="text"
      style={
        Object {
          "color": "hsl(356, 3%, 59%)",
          "fontFamily": "Inter",
          "fontSize": "14px",
          "fontWeight": 400,
          "textAlign": "initial",
          "textTransform": "initial",
        }
      }
    >
      Some Text
    </span>
  </Component>
</Typography>
`;
